genesis/streaming.py

```diff
@@ -14,350 +14,414 @@
+            else:
+                # unpack messages and check for errors
+                yield list(self._validate_and_process(msgs))
+
+                # reset the timeout counter
+                last_msg_time = time.time()
+
+            # actualy commit any offsets the user committed
+            self._commit_to_kafka(defer=True)
+
+    # returns a generator returning deserialized, user-processed messages
+    def _filtered_stream(self, mapper, filter, timeout):
+        t_last = time.time()
+
+        for msgs in itertools.chain(
+            [self._buffer.values()], self._raw_stream(timeout=timeout)
+        ):
+            if not is_heartbeat(msgs):
+                self._buffer = dict(enumerate(msgs))
+
+                # process the messages on the workers
+                for i, (rec, meta) in enumerate(
+                    mapper(ParseAndFilter(parser=self._parser, filter=filter), msgs)
+                ):
+                    # pop the message from the buffer, indicating we've processed it
+                    del self._buffer[i]
+
+                    # mark as consumed and increment the index _before_ we yield,
+                    # as we may never come back from yielding (if the user decides
+                    # to break from the loop)
+                    self._consumed[(meta.topic, meta.partition)] = meta.offset
+
+                    # yield if the filter didn't return None
+                    if rec is not None:
+                        yield (rec, meta)
+
+    def commit(self, defer=True):
+        self._committed = self._consumed.copy()
+        self._commit_to_kafka(defer=defer)
+
+    def _commit_to_kafka(self, defer=True):
+        # Occasionally commit read offsets (note: while this ensures we won't lose any
+        # messages, some may be duplicated if the program is killed before offsets are committed).
+        now = time.time()
+        last = self.last_commit_time if self.last_commit_time is not None else 0
+        if self._committed and (not defer or now - last > 5.0):
+            print("COMMITTING", file=sys.stderr)
+            tp = [
+                TopicPartition(_topic, _part, _offs + 1)
+                for (_topic, _part), _offs in self._committed.items()
+            ]
+            # print("   TP=", tp)
+            self.c.commit(offsets=tp)
+
+            # drop all _committd offsets from _consumed; no need to commit them again if the user
+            # calls commit()
+            self._consumed = {
+                k: v for k, v in self._consumed.items() if k not in self._committed
+            }
+            self._committed = {}
+
+            self.last_commit_time = now
+
+    def _stream(self, filter, mapper, progress, timeout, limit):
+        import warnings
+
+        with warnings.catch_warnings():
+            # hide the annoying 'TqdmExperimentalWarning' warning
+            warnings.simplefilter("ignore")
+            from tqdm.autonotebook import tqdm
+
+        t = tqdm(
```

Comment:
Similar to one of the other comments above, something like:
```
foo = long_function_name(var_one, var_two,
                         var_three, var_four)
```
is a more common Python style (& PEP8 recommended).